src/current/v23.1/changefeed-messages.md

```diff
@@ -23,35 +23,103 @@ This page describes the format and behavior of changefeed messages. You will fin
+
+{{site.data.alerts.callout_info}}
+Changefeeds created with [`EXPERIMENTAL CHANGEFEED FOR`](changefeed-for.html) or [`CREATE CHANGEFEED`](create-changefeed.html) with no sink specified (sinkless changefeeds) produce messages without the envelope metadata fields of changefeeds emitting to sinks.
+{{site.data.alerts.end}}
+
+The following sections provide examples of changefeed messages that are emitted when you specify each of the supported `envelope` options. Other [changefeed options](create-changefeed.html#options) can affect the message envelope and the what messages are emitted. Therefore, the examples are a guide for what you can expect when only the `envelope` option is specified.
+
+### `wrapped`
+
+`wrapped` is the default envelope structure for changefeed messages. This envelope contains an array of the primary key, a top-level field for the type of message, and the current state of the row (or `null` for [deleted rows](#delete-messages)). Kafka sinks do not contain an array of the primary key because the key is part of the message metadata.
+
+- Cloud storage sink:
+
+    ~~~
+    {"after": {"city": "seattle", "creation_time": "2019-01-02T03:04:05", "current_location": "86359 Jeffrey Ranch", "ext": {"color": "yellow"}, "id": "68ee1f95-3137-48e2-8ce3-34ac2d18c7c8", "owner_id": "570a3d70-a3d7-4c00-8000-000000000011", "status": "in_use", "type": "scooter"}, "key": ["seattle", "68ee1f95-3137-48e2-8ce3-34ac2d18c7c8"]}
+    ~~~
+
+- Kafka sink:
+
+    ~~~
+    {"after": {"city": "washington dc", "creation_time": "2019-01-02T03:04:05", "current_location": "24315 Elizabeth Mountains", "ext": {"color": "yellow"}, "id": "dadc1c0b-30f0-4c8b-bd16-046c8612bbea", "owner_id": "034075b6-5380-4996-a267-5a129781f4d3", "status": "in_use", "type": "scooter"}}
+    ~~~
+
+### `bare`
+
+`bare` removes the `after` key from the changefeed message and stores any metadata in a `crdb` field. When used with [`avro`](#avro) format, `record` will replace the `after` key. Note that `bare` is the default envelope format for [CDC queries](cdc-queries.html). For an example, refer to [Filter columns](cdc-queries.html#filter-columns). While `bare` is compatible with all sinks, a message emitted to Kafka will not contain the `crdb` field; the message envelope will be identical to the [`row`](#row) envelope option.
+
+- Cloud storage sink:
+
+    ~~~
+    {"__crdb__": {"key": ["washington dc", "cd48e501-e86d-4019-9923-2fc9a964b264"]}, "city": "washington dc", "creation_time": "2019-01-02T03:04:05", "current_location": "87247 Diane Park", "ext": {"brand": "Fuji", "color": "yellow"}, "id": "cd48e501-e86d-4019-9923-2fc9a964b264", "owner_id": "a616ce61-ade4-43d2-9aab-0e3b24a9aa9a", "status": "available", "type": "bike"}
+    ~~~
+
+### `key_only`
+
+`key_only` emits only the key and no value, which is faster if you only need to know the key of the changed row. This envelope option is only supported for [Kafka sinks](changefeed-sinks.html#kafka) or sinkless changefeeds.
```

Comment:
Should be supported for gcpubsub too.

Reply:
I tested this and was getting an incompatible sink error with Pub/Sub?

Reply:
Ack, sorry. I believe you but I'm not sure why it's not supported.

Reply:
Oh yeah, honestly I tried again, because I worried I misremembered. I was testing on a serverless cluster to gcpubsub — should I write up an issue for this?

Reply:
It's very possible we made this decision for a good reason and I'm just misremembering, sorry. Don't think we need an issue for it, we'll just revisit it if we need other options for some reason.